kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -266,20 +266,14 @@ public void testGetBestSolution() throws Exception {
         ServiceResponse<SolverInstance> response = solverClient.updateSolverState( CONTAINER_1_ID, SOLVER_1_ID, instance );
         assertSuccess( response );
         assertEquals( SolverInstance.SolverStatus.SOLVING, response.getResult().getStatus() );
-
-        Solution solution = null;
+        // let it run for 5 sec to produce any solution
         // It can take a while for the Construction Heuristic to initialize the solution
-        for (int i = 0; i < 15; i++) {
-            ServiceResponse<SolverInstance> solutionResponse = solverClient.getSolverBestSolution(CONTAINER_1_ID, SOLVER_1_ID);
-            assertSuccess(solutionResponse);
-            solution = solutionResponse.getResult().getBestSolution();
-            // Only once the solution's score is null, the solution is fully uninitialized
-            // TODO add "|| solution.getScore().isInitialized()" once PLANNER-405 is fixed
-            if (solution != null && solution.getScore() != null) {
-                break;
-            }
-            Thread.sleep(1000);
-        }
+        Thread.sleep(5000);
```

Comment:
5000ms is both too much and too little. On a fast machine it unnecessarily slows down the test by 5 seconds (that's a lot, because 3 formats make 15secodns - if other tests would have this same philosophy, it would take so long to run all tests, that no one runs all tests before committing). On slow machines, it can fail with a false negative. The original code is intended to take 1000ms on a fast machine and 15000ms on a really slow one (and I believe we have to increase the loop to 60, so on really really slow machine 60000ms is still acceptable - well actually we have to make it an infinite loop and set a @Test.timeout of 60 seconds - I 'll send a PR later)

The fact that this change fixes the problem, means that something is not adhering to the contract. The contract states that the solution.getScore() is null as long as the solution isn't initialized. Furthermore, I believe the solution is null if the solver hasn't started yet. I suspect **the race condition we are seeing is because we're getting the solution of the previous run, if the solver hasn't started yet, instead of null.**